src/test/java/com/commercetools/pspadapter/notification/processor/impl/PaymentSalePendingProcessorTest.java

```diff
@@ -0,0 +1,84 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static com.commercetools.testUtil.CompletionStageUtil.executeBlocking;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PaymentSalePendingProcessorTest {
+
+    @Test
+    public void shouldCallUpdatePaymentWithCorrectArgs() {
+        // set up
+        Payment ctpMockPayment = mock(Payment.class);
+        Transaction transaction = mock(Transaction.class);
+        when(ctpMockPayment.getTransactions()).thenReturn(Collections.singletonList(transaction));
+        when(transaction.getType()).thenReturn(TransactionType.CHARGE);
+        when(transaction.getState()).thenReturn(TransactionState.SUCCESS);
+
+        NotificationProcessorBase processorBase = spy(new PaymentSalePendingProcessor(new GsonBuilder().create()));
+
+        doReturn(CompletableFuture.completedFuture(Optional.of(ctpMockPayment)))
+                .when(processorBase).getRelatedCtpPayment(any(), any());
+        SphereClient sphereClient = mock(SphereClient.class);
+
+        PaymentService paymentService = spy(new PaymentServiceImpl(sphereClient));
+
+        CtpFacade ctpFacade = spy(
+                new CtpFacade(mock(CartService.class), mock(OrderService.class), paymentService)
+        );
+
+        Event event = new Event();
+        event.setEventType(NotificationEventType.PAYMENT_SALE_PENDING.toString());
+
+        // test
+        doAnswer(invocation -> {
+            verifyUpdatePaymentCall(ctpMockPayment, invocation);
+            return CompletableFuture.completedFuture(ctpMockPayment);
+        }).when(paymentService).updatePayment(any(Payment.class), anyList());
+
+        Payment returnedPayment = executeBlocking(processorBase.processEventNotification(ctpFacade, event));
+
+        // assert
+        assertThat(returnedPayment).isEqualTo(ctpMockPayment);
+        verify(paymentService, times(1))
+                .updatePayment(any(Payment.class), anyList());
+    }
+
+    private void verifyUpdatePaymentCall(Payment ctpMockPayment, InvocationOnMock invocation) {
+        Payment payment = invocation.getArgumentAt(0, Payment.class);
+        List<UpdateAction<Payment>> updateActions = invocation.getArgumentAt(1, List.class);
+        assertThat(payment).isEqualTo(ctpMockPayment);
+        assertThat(updateActions.size()).isEqualTo(2);
+        ChangeTransactionState changeTransactionState = (ChangeTransactionState) updateActions.get(1);
+        assertThat(changeTransactionState.getState()).isEqualTo(TransactionState.PENDING);
```

Comment:
2 update actions are expected, but only one verified. Is it on purpose?

Reply:
It's on purpose.

Reply:
so what is the purpose? why one items verification is skipped?

Reply:
The other one is AddInterfaceInteraction, which is common for all notification processors. I already covered this case in other test, so it's not necessary to repeat it here.

Reply:
ok, then put is as a comment close to `updateActions.get(1);` line, otherwise 1 looks like a magic number.

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/69/commits/2777b7c9b71cafb9cb9c221688af5044ed243af1